### PR TITLE
TST: Some fixes & refactoring around glibc-dependent skips in test_umath.py

### LINKED ISSUE #15179: TestAVXFloat32Transcendental test failure
This test is failing on 64-bit Linux on the conda-forge Azure CI when building NumPy 1.18.0, see https://github.com/conda-forge/numpy-feedstock/pull/179. Failed on Python 3.6, 3.7 and 3.8, so it's consistent. Not sure what's different about that setup, could be glibc version, compiler, or some Azure - TravisCI delta.

```
=================================== FAILURES ===================================
_______________ TestAVXFloat32Transcendental.test_sincos_float32 _______________

self = <numpy.core.tests.test_umath.TestAVXFloat32Transcendental object at 0x7f6a5342e080>

    def test_sincos_float32(self):
        np.random.seed(42)
        N = 1000000
        M = np.int_(N/20)
        index = np.random.randint(low=0, high=N, size=M)
        x_f32 = np.float32(np.random.uniform(low=-100.,high=100.,size=N))
        # test coverage for elements > 117435.992f for which glibc is used
        x_f32[index] = np.float32(10E+10*np.random.rand(M))
        x_f64 = np.float64(x_f32)
>       assert_array_max_ulp(np.sin(x_f32), np.float32(np.sin(x_f64)), maxulp=2)
E       AssertionError: Arrays are not almost equal up to 2 ULP
```

This functionality came in in gh-13368, new in 1.18.0. Likely the accuracy is just slightly over 2 ULP, conda-forge tests need to be re-run with larger tolerance. (related: PR for improving the test function to show actual difference in gh-15178)

Cc @r-devulap

### PR DESCRIPTION
**Pull request by h-vetinari**
I was looking at the test [skips](https://github.com/conda-forge/numpy-feedstock/blob/99ba8ef9cd600d3fbeca27360b58abe90c66a890/recipe/meta.yaml#L40-L44) in the numpy feedstock by chance, and thought of fixing #15179, now that there's a glibc-version check that was introduced in #19209.

First step is taking that check and turning it into a reusable boolean. The second commit applies that also to a different glibc-relevant test ~(though, strictly speaking, it's left open in https://github.com/numpy/numpy/commit/8b2b7a2a5549e33f2d7187c4f78de69a8c482c83 whether this also works for 2.17 & 2.18 - I can parametrize the check of course, but not sure that's worth the effort?)~

Finally, applying the fix [suggested](https://github.com/numpy/numpy/issues/15179#issuecomment-839070581) by @r-devulap (which has been tested in a conda-forge PR to work) conditionally on the glibc-version, so that the test in the numpy-feedstock can be switched on unconditionally.

Fixes #15179

### CODE REVIEW

**Comment by h-vetinari** [comment]
> The second commit applies that also to a different glibc-relevant test (though, strictly speaking, it's left open in [8b2b7a2](https://github.com/numpy/numpy/commit/8b2b7a2a5549e33f2d7187c4f78de69a8c482c83) whether this also works for 2.17 & 2.18 - I can parametrize the check of course, but not sure that's worth the effort?)

I decided to fix this, because the correct inclusion/exclusion was actually off for the glibc-version in manylinux 2014.

**Comment by h-vetinari** [comment]
Not sure what the timeline for 1.22 is, but it would be good if this PR could be looked at before that (and merged, ideally, as it would allow conda-forge to remove a test skip - which are hardly ever looked at and prone to bit rot).

CC @charris @seberg @rgommers

File: numpy/core/tests/test_umath.py [hunk 1]
```diff
@@ -28,9 +28,7 @@ def get_glibc_version():
 
 
 glibcver = get_glibc_version()
-glibc_newerthan_2_17 = pytest.mark.xfail(
-        glibcver != '0.0' and glibcver < '2.17',
-        reason="Older glibc versions may not raise appropriate FP exceptions")
+glibc_older_than = lambda x: (glibcver != '0.0' and glibcver < x)
```
Comment (charris) [inline numpy/core/tests/test_umath.py:32]: The use of alphabetical order works, but it bothers me :)

File: numpy/core/tests/test_umath.py [hunk 2]
```diff
@@ -3439,13 +3433,14 @@ def check(x, rtol):
         x_series = np.logspace(-20, -3.001, 200)
         x_basic = np.logspace(-2.999, 0, 10, endpoint=False)
 
-        if dtype is np.longcomplex:
+        if glibc_older_than("2.19") and dtype is np.longcomplex:
```
Comment (charris) [inline numpy/core/tests/test_umath.py:3436]: "2.19" or "2.18"?

File: numpy/core/tests/test_umath.py (hunk 1 above)
Comment (h-vetinari) [reply numpy/core/tests/test_umath.py:32]: I tend to agree, but then, it was this way before already, and I didn't want to overengineer something for 3 tests...

What would you prefer? Comparing tuples? Using something like `distutils.version.LooseVersion`?

File: numpy/core/tests/test_umath.py (hunk 2 above)
Comment (h-vetinari) [reply numpy/core/tests/test_umath.py:3436]: Not sure what that would do? If it's older than 2.18, ít's definitely older than 2.19?

Comment (charris) [reply numpy/core/tests/test_umath.py:3436]: I was looking at the older comment that it was OK for glibc > 1.17.

Comment (charris) [reply numpy/core/tests/test_umath.py:3436]:
```
    skip_longcomplex_msg = ('Trig functions of np.longcomplex values known to be '
                            'inaccurate on aarch64 for some compilation '
                            'configurations, should be fixed by building on a '
                            'platform using glibc>2.17')
```

Comment (h-vetinari) [reply numpy/core/tests/test_umath.py:3436]: For the if-branches here, the pertinent part is:
```
            # Can use 2.1 for > Ubuntu LTS Trusty (2014), glibc = 2.19.
            if skip_longcomplex_msg:
                pytest.skip(skip_longcomplex_msg)
```
I take from this comment that anything from glibc 2.19 onwards should be fine to take the `else`-clause (with the `2.1*eps`). I cannot very (easily or quickly) how far back in terms of glibc-versions things would work.

Comment (h-vetinari) [reply numpy/core/tests/test_umath.py:3436]: > `[...] should be fixed by building on a platform using glibc>2.17'`

Right I couldn't evaluate that corner case (2.18), so I went with the conservative choice, of still running the check (resp. the looser tolerance) with 2.18.

Before the last commit, I actually had this as `glibc_older_than_2_17`, but then realized that this would do the wrong thing (due to `glibc > 2.17`) for manylinux 2014.

**Comment by h-vetinari** [comment]
Thanks a lot for taking the time to review @charris! :)

File: numpy/core/tests/test_umath.py (hunk 2 above)
Comment (charris) [reply numpy/core/tests/test_umath.py:3436]: OK, probably not worth worrying about. Manylinux2014 itself is getting toward end of life.

**Comment by charris** [comment]
Thanks @h-vetinari .
